### DISCUSSION TITLE
Server build fails

### DISCUSSION THREAD

**Discussion by dsrdjevic** [community/community#139054] [Actions] [answered]
### Select Topic Area

General

### Body

I am not able to resolve this issue. Documentatioin wasn't helpfull for me. Could somebody who experienced the similar problem take a look? 
Thanks

My server build fails with the follwoing warning message:
The following actions use a deprecated Node.js version and will be forced to run on node20: actions/setup-node@v3.

Here is my workflow file main-XXX-server.yml:

name: Build and deploy Node.js app to Azure Web App - dedaai-server

on:
  push:
    branches:
      - main
  workflow_dispatch:

jobs:
  build:
    runs-on: windows-latest

    steps:
      - uses: actions/checkout@v4

      - name: Set up Node.js version
        uses: actions/setup-node@v3
        with:
          node-version: '20.x'

      - name: npm install, build, and test
        run: |
          npm install
          npm run build --if-present
          npm run test --if-present
      
      - name: Upload artifact for deployment job
        uses: actions/upload-artifact@v4
        with:
          name: node-app
          path: .

  deploy:
    runs-on: ubuntu-latest
    needs: build
    environment:
      name: 'Production'
      url: ${{ steps.deploy-to-webapp.outputs.webapp-url }}

    steps:
      - name: Download artifact from build job
        uses: actions/download-artifact@v4
        with:
          name: node-app

      - name: 'Deploy to Azure Web App'
        uses: azure/webapps-deploy@v2
        id: deploy-to-webapp
        with:
          app-name: 'dedaai-server'
          slot-name: 'Production'
          publish-profile: ${{ secrets.AZUREAPPSERVICE_PUBLISHPROFILE_C67A4A2332FC400A98E4C5FFECB4FAC8 }}
          package: .

**Comment by JanneKiiskila** [comment]
I think that is just a warning, look at the logs for the actual error.
You can get rid of that warning by updating your version of `actions/setup-node@v3` to `actions/setup-node@v4`.

**Reply by dsrdjevic** [reply]
Log shows this error:
rror: EMFILE: too many open files, open 'D:\a\DedaAIServer\DedaAIServer\node_modules\next\dist\compiled\schema-utils2\index.js'

**Reply by JanneKiiskila** [reply]
That is the real problem you need to solve.

**Reply by JanneKiiskila** [reply, marked as answer]
https://stackoverflow.com/questions/38099671/node-js-on-windows-frequently-fails-with-emfile-too-many-open-files will probably give you some clues.

**Reply by dsrdjevic** [reply]
Thanks for help.
It seems that my problem started with changing originally actions/setup-node@v2 to actions/setup-node@v4.

I'll continue digging.

**Reply by JanneKiiskila** [reply]
Do you have to build in Windows? Probably easier to solve on Linux builders.

**Reply by dsrdjevic** [reply]
I tried Github to build on Linux and it succeeded!
But, now build deploy failed. It seems I reached the limit of 1GB of free space. It is strange because it is my firts build on Linux VM and I assume that this disk space is not the same as space form Windows VM.

**Reply by JanneKiiskila** [reply]
There is more than 1 GB free space. GitHub Linux runners usually have ~15...20 GB free space, but of course the space is then consumed by your actions. Use `df -h` to check how much space you have between steps.

If you see you are running out - delete some temporary files in-between steps or use a local self-hosted runner with more disk space.

**Comment by anurag-iitk** [comment]
To resolve the issue, update your workflow file to use Node.js 20.x by changing the node-version under setup-node to '20.x'

**Reply by dsrdjevic** [reply]
This is my workflow with changes, but still the same issue:
name: Build and deploy Node.js app to Azure Web App - dedaai-server

on:
  push:
    branches:
      - main
  workflow_dispatch:

jobs:
  build:
    runs-on: windows-latest

    steps:
      - uses: actions/checkout@v4

      - name: Set up Node.js version
        uses: actions/setup-node@v4
        with:
          node-version: '20.x'

      - name: npm install, build, and test
        run: |
          npm install
          npm run build --if-present
          npm run test --if-present
      
      - name: Upload artifact for deployment job
        uses: actions/upload-artifact@v4
        with:
          name: node-app
          path: .

  deploy:
    runs-on: ubuntu-latest
    needs: build
    environment:
      name: 'Production'
      url: ${{ steps.deploy-to-webapp.outputs.webapp-url }}

    steps:
      - name: Download artifact from build job
        uses: actions/download-artifact@v4
        with:
          name: node-app

      - name: 'Deploy to Azure Web App'
        uses: azure/webapps-deploy@v2
        id: deploy-to-webapp
        with:
          app-name: 'dedaai-server'
          slot-name: 'Production'
          publish-profile: ${{ secrets.AZUREAPPSERVICE_PUBLISHPROFILE_C67A4A2332FC400A98E4C5FFECB4FAC8 }}
          package: .

**Reply by dsrdjevic** [reply]
It seems that I resolved "Too many open files.." issue.  It builds now on windows-latest with no error.It might be a combination of the following:
 1) I restored v2 from v4:
      - uses: actions/checkout@v2
  2) changed path: . to path: build/ at build: with: path
  3) increased the storage space
  4) my account was reset on Oct 1.

Now, I encounter a new error in YML file. This error appeared when VS Code asked me to upgrade (or install) Github actions
Error is in above file at line deploy: environment: name: 'Production'.  It does not like 'Production" any more and it seems it expect env variable.
Any suggestion how to resolve it?
